### PR TITLE
Feature/dd 11849

### PR DESCRIPTION
**Pull request by timmixell**

### CODE REVIEW

File: Cache.cfc [hunk 1]
```diff
@@ -0,0 +1,367 @@
+component extends = "lib.util.EhcacheContainer" implements = "lib.sql.IQueryable" {
+
+	Cache function init(required string name, string managerName) {
+		return super.init(argumentCollection = arguments);
+	}
+
+	any function attributesFor(required any element) {
+		local.s = createObject("java", "java.util.HashMap").init();
```
Comment (andrewskaggs83) [inline Cache.cfc]: @timmixell Is there a compelling reason for this to be a single letter variable vs something more descriptive?

Comment (timmixell) [reply Cache.cfc]: not really - i can replace it if you have a real name for it :D

Comment (dsbrady) [reply Cache.cfc]: local.hashMap?

File: Cache.cfc [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+		local.v = element.getValue();
```
Comment (dsbrady) [inline Cache.cfc]: local.value or local.elementValue?

File: Cache.cfc [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+		return variables.queryable.fieldIsFilterable(arguments.fieldName);
+	}
+
+	string function getFieldList() {
+		queryableCheck();
+
+		return listReduce(
+			variables.queryable.getFieldList(),
+			function(v, i) {
```
Comment (dsbrady) [inline Cache.cfc:283]: @timmixell Not sure if we want more descriptive variable names here or not

File: Manager.cfc [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+		return getInstance().cacheExists(arguments.name);
+	}
+
+	Manager function createFromPath(required string path) {
+		return new Manager().setInstance(createObject("java", "net.sf.ehcache.CacheManager").newInstance(arguments.path));
+	}
+
+	Manager function createFromXML(required string xml) {
+		local.is = createObject("java", "java.io.ByteArrayInputStream").init(arguments.xml.getBytes());
```
Comment (dsbrady) [inline Manager.cfc:46]: @timmixell I could go either way on this one. We allow acronyms, but this also looks like "is" so ... :shurg:

File: tests/TestManager.cfc [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+		try {
+			variables.cacheManager = new lib.ehcache.Manager();
+		} catch(Any e) {
+			variables.exception = e;
+		}
+	}
+
+	function afterTests() {
+		variables.cacheManager.getCache("mxunitCache").flush();
+//		variables.cacheManager.removeCache("mxunitCache");
```
Comment (dsbrady) [inline tests/TestManager.cfc:13]: @timmixell do we need these commented lines anymore? Or keep them for later use?

File: Cache.cfc (hunk 1 above)
Comment (timmixell) [reply Cache.cfc]: attributesFor internal vars have been updated

File: Cache.cfc (hunk 3 above)
Comment (timmixell) [reply Cache.cfc:283]: i'm gonna draw the line here - i would compare this to declaring for-loop vars

**Review by dsbrady — APPROVED** [review]

**Review by kishore12 — APPROVED** [review]
looks good to me

**Pull request merged**